cmd/thanos/compact.go

```diff
@@ -41,6 +42,10 @@ func registerCompact(m map[string]setupFunc, app *kingpin.Application, name stri
 
 	retention := modelDuration(cmd.Flag("retention.default", "How long to retain samples in bucket. 0d - disables retention").Default("0d"))
```

Comment:
Can we maybe remove this flag then? (: It's bad to have multiple ways of doing same thing. What do you think?

Reply:
(I am aware initially we though we can compose this, but I feel like this flag is not required)

Reply:
:+1: as I understand it setting these new 3 flags to one interval would be the same thing, right as `retention.default`?

Reply:
yes (: That's my thinking

Reply:
Yes it should be exactly the same :-)